### PR TITLE
EOS-19858: utils_setup should log its activity

### PR DESCRIPTION
**Pull request by vastradparayya**
Signed-off-by: Parayya Vastrad <parayya.vastrad@seagate.com>

# Problem Statement
- utils_setup should log its activity

# Design
-  JIRA - [EOS-19858](https://jts.seagate.com/browse/EOS-19858)

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: NA
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: NA
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  Y

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Generic comment / observation:
Log.info should be used to describe the status like post_install passed. Rest all should be changed as Log.debug. By default there should be minimum logging and that should be useful.
Please make corrections likewise.

File: py-utils/src/setup/utils.py [hunk 1]
```diff
@@ -44,36 +50,35 @@ class Utils:
...
 
     @staticmethod
     def _get_kafka_server_list(conf_url: str):
         """ Reads the ConfStore and derives keys related to message bus """
-
-        Conf.load('cluster_config', conf_url)
-
-        key_list = ['cortx>software>common>message_bus_type', \
-                    'cortx>software>kafka>servers']
+        Conf.load("cluster_config", conf_url)
```
Comment (selvakumaar5496) [inline py-utils/src/setup/utils.py:72]: single quote

File: py-utils/src/setup/utils.py [hunk 2]
```diff
@@ -192,42 +219,66 @@ def test():
...
         msg_test.send_msg(["Test Message"])
         # Recieve the same & validate
+        Log.debug("Sent message 'Test Message' using MessageBusTest's" +\
+            "send_msg")
         msg = msg_test.receive_msg()
-        if str(msg.decode('utf-8')) != "Test Message":
-            raise SetupError(errno.EINVAL, "Unable to test the config. Received \
+        Log.debug("Validating the sent(Test Message) and received" +\
+            "(Test Message) are same")
+        if str(msg.decode("utf-8")) != "Test Message":
```
Comment (selvakumaar5496) [inline py-utils/src/setup/utils.py:227]: ('utf-8')

File: py-utils/src/setup/utils.py [hunk 3]
```diff
@@ -44,36 +50,35 @@ class Utils:
...
         Conf.save('index')
         # copy this conf file as message_bus.conf
         try:
-            os.rename('/etc/cortx/message_bus.conf.sample', \
-                      '/etc/cortx/message_bus.conf')
+            os.rename(f'/{conf_file}.sample', conf_file)
         except OSError as e:
-            raise SetupError(e.errno, "Failed to create \
-                /etc/cortx/message_bus.conf %s", e)
+            Log.exception(f"Failed to create {conf_file} - {e}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:66]: No need to log here. Exception logging can be at one place where SetupError is caught.

File: py-utils/src/setup/utils_setup.py [hunk 4]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
+            "doing utils cleanup now")
         rc = Utils.cleanup()
+        Log.info(f"Utils cleanup was successful with rc {rc}")
         return rc
 
 
 def main(argv: dict):
     try:
         desc = "CORTX Utils Setup command"
+        Log.debug(f"Received {argv[1]} command to run")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:196]: Message should be professional. 
Log.info(f"Starting utils_setup {argv[1]}...")

File: py-utils/src/setup/utils_setup.py [hunk 5]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
         command = Cmd.get_command(desc, argv[1:])
         rc = command.process()
+        Log.debug(f"Executed {command}.process")
         if rc != 0:
+            Log.error(f"Could not able to run {argv[1]}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:201]: Incorrect grammar and place 
```
rc = command.process()
Log.info(f"Command {command} {argv[1]} finished with exit code {rc}")
sys.exit(rc)
```

File: py-utils/src/setup/utils_setup.py [hunk 6]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
             raise ValueError(f"Failed to run {argv[1]}")
     except Exception as e:
+        Log.exception(f"Failed to run {argv[1]} - {e}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:204]:
```
except SetupError as e:
    sys.stderr.write("error: %s\n\n" % str(e))
    sys.stderr.write("%s\n" % traceback.format_exc())
    Cmd.usage(argv[0])
except SetupError as e:
    sys.stderr.write("%s\n" % traceback.format_exc())
```

File: py-utils/src/setup/utils_setup.py [hunk 7]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
 
     def __init__(self, args: dict):
         super().__init__(args)
 
     def process(self):
         Utils.validate('cleanup')
+        Log.debug("Validated cleanup using Utils.validate('cleanup') and " \
+            "doing utils cleanup now")
         rc = Utils.cleanup()
+        Log.info(f"Utils cleanup was successful with rc {rc}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:189]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils_setup.py [hunk 8]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ Cleanup Setup Cmd """
-    name = "cleanup"
+    name = 'cleanup'
 
     def __init__(self, args: dict):
         super().__init__(args)
 
     def process(self):
         Utils.validate('cleanup')
+        Log.debug("Validated cleanup using Utils.validate('cleanup') and " \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:186]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 9]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ Reset Setup Cmd """
-    name = "reset"
+    name = 'reset'
 
     def __init__(self, args):
         super().__init__(args)
 
     def process(self):
         Utils.validate('reset')
+        Log.debug("Validated reset using Utils.validate('reset') and doing " \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:170]: Remove this log. Not required.

File: py-utils/src/setup/utils_setup.py [hunk 10]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ Test Setup Cmd """
-    name = "test"
+    name = 'test'
 
     def __init__(self, args):
         super().__init__(args)
 
     def process(self):
         Utils.validate('test')
+        Log.debug("Validated test using Utils.validate('test') and doing " \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:154]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 11]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ Init Setup Cmd """
-    name = "init"
+    name = 'init'
 
     def __init__(self, args):
         super().__init__(args)
 
     def process(self):
         Utils.validate('init')
+        Log.debug("Validated init using Utils.validate('init') and doing "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:138]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 12]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ Setup Config Cmd """
-    name = "config"
+    name = 'config'
 
     def __init__(self, args):
         super().__init__(args)
 
     def process(self):
         Utils.validate('config')
+        Log.debug("Validated config and doing utils config now")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:123]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 13]
```diff
@@ -72,105 +80,128 @@ def add_args(parser: str, cls: str, name: str):
...
     """ PostInstall Setup Cmd """
-    name = "post_install"
+    name = 'post_install'
 
     def __init__(self, args: dict):
         super().__init__(args)
 
     def process(self):
         Utils.validate('post_install')
+        Log.debug("Validated post_install and doing utils post_install" +\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:95]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 14]
```diff
@@ -54,14 +57,19 @@ def get_command(desc: str, argv: dict):
         """ Return the Command after parsing the command line. """
 
         parser = argparse.ArgumentParser(desc)
-
+        Log.debug(f"parsed description {desc}")
         subparsers = parser.add_subparsers()
         cmds = inspect.getmembers(sys.modules[__name__])
+        Log.debug(f"Listed all members {sys.modules[__name__]}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:63]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 15]
```diff
@@ -54,14 +57,19 @@ def get_command(desc: str, argv: dict):
...
         cmds = [(x, y) for x, y in cmds
-            if x.endswith("Cmd") and x != "Cmd"]
+            if x.endswith('Cmd') and x != 'Cmd']
+        Log.debug(f"Listed filtered classes of phases - {cmds}")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:66]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 16]
```diff
@@ -54,14 +57,19 @@ def get_command(desc: str, argv: dict):
...
         for name, cmd in cmds:
             cmd.add_args(subparsers, cmd, name)
+        Log.debug(f"Parsing command line {argv} for returning command")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:69]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils_setup.py [hunk 17]
```diff
@@ -54,14 +57,19 @@ def get_command(desc: str, argv: dict):
...
         args = parser.parse_args(argv)
+        Log.debug(f"Parsed command line {argv} and returning cmd "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:71]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils.py [hunk 18]
```diff
@@ -136,20 +142,31 @@ def post_install():
...
+            Log.debug("Checking whether required package is installed ")
             for package in enumerate(packages):
                 if result.find(package[1][:-1]) == -1:
-                    raise SetupError(errno.EINVAL, "Required python package %s \
-                        is missing", package[1][:-1])
+                    Log.error(f"Required {package[1][:-1]} package is not"\
+                        f" installed or missing")
+                    raise SetupError(errno.EINVAL, "Required python " +\
+                        "package %s is missing" % package[1][:-1])
+        Log.info("Utils post_install phase successful - All python packages"\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:168]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 19]
```diff
@@ -161,27 +178,37 @@ def init():
         try:
             admin = MessageBusAdmin(admin_id='register')
             admin.register_message_type(message_types=['IEM'], partitions=1)
+            Log.debug("Created message_type 'IEM' using message bus admin")
         except MessageBusError as e:
+            Log.exception(f"Failed to create message_type 'IEM' - {e}")
             raise SetupError(e.rc, "Unable to create message_type. %s", e)
+        Log.info("Utils init passed successfully")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:185]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 20]
```diff
@@ -161,27 +178,37 @@ def init():
...
         server_info = Utils._get_server_info(conf_url, Conf.machine_id)
+        Log.debug(f"Received server info. {server_info}")
         if server_info is None:
-            raise SetupError(errno.EINVAL, "Could not find server information \
-                in %s", conf_url)
+            Log.error(f"Could not find server information in {conf_url}")
+            raise SetupError(errno.EINVAL, "Could not find server " +\
+                "information in %s", conf_url)
         Utils._create_cluster_config(server_info)
+        Log.info("Utils config phase done successfully")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:211]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 21]
```diff
@@ -192,42 +219,66 @@ def test():
...
-        if str(msg.decode('utf-8')) != "Test Message":
-            raise SetupError(errno.EINVAL, "Unable to test the config. Received \
+        Log.debug("Validating the sent(Test Message) and received" +\
+            "(Test Message) are same")
+        if str(msg.decode("utf-8")) != "Test Message":
+            Log.error("Could not test config phase of utils as sent and " +\
+                "received message are not matching")
+            raise SetupError(errno.EINVAL, "Unable to test the config. Received\
                 message is %s", msg)
+        Log.info("Utils test phase was successful")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:232]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 22]
```diff
@@ -192,42 +219,66 @@ def test():
...
                 raise SetupError(e.rc, "Can not reset Message Bus. %s", e)
             except Exception as e:
-                raise SetupError(errors.ERR_OP_FAILED, "Can not reset Message Bus. \
-                    %s", e)
+                Log.exception(f"Error while deleting message_types {e}")
+                raise SetupError(errors.ERR_OP_FAILED, "Can not reset Message  \
+                    Bus. %s", e)
+        else:
+            Log.error(f"Conf file {conf_file} does not exist!")
+        Log.info("Utils reset was successful")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:260]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 23]
```diff
@@ -192,42 +219,66 @@ def test():
...
                     os.remove(each_file)
+                    Log.debug(f"Deleted conf file {each_file}")
                 except OSError as e:
+                    Log.exception(f"Error while deleting config file " \
+                        f"{each_file} {e}")
                     raise SetupError(e.errno, "Error deleting config file %s, \
                         %s", each_file, e)
+            else:
+                Log.error(f"Conf file {each_file} does not exist")
+        Log.info("utils cleanup was successful")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:283]: No need to print info level. It should only be at the main entry point and exit point. Remove this, its anyway going to print this message.

File: py-utils/src/setup/utils.py [hunk 24]
```diff
@@ -89,16 +94,17 @@ def _get_kafka_server_list(conf_url: str):
                 port_list.append(all_servers[i][rc + 1:])
                 kafka_server_list.append(all_servers[i][:rc])
         if len(kafka_server_list) == 0:
-            raise SetupError(errno.EINVAL, "No valid Kafka server info \
-                provided for config key: %s", key_list[1])
+            Log.error(f"No valid Kafka server info provided for Config Key "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:97]: log.error(f"Missing config entry {key_list} in config file {conf_file")

File: py-utils/src/setup/utils.py [hunk 25]
```diff
@@ -108,19 +114,19 @@ def _get_server_info(conf_url: str, machine_id: str) -> dict:
...
-        Conf.load('cluster', 'json:///etc/cortx/cluster.conf.sample')
+        Conf.load('cluster', f'json://{conf_file}.sample')
         for key, value in server_info.items():
             Conf.set('cluster', f'server_node>{key}', value)
         Conf.save('cluster')
         # copy this conf file as cluster.conf
         try:
-            os.rename('/etc/cortx/cluster.conf.sample', \
-                '/etc/cortx/cluster.conf')
+            os.rename(f'{conf_file}.sample', conf_file)
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:126]: We dont want to overwrite if conf file exists already. We need to merge to it.

File: py-utils/src/setup/utils.py [hunk 26]
```diff
@@ -136,20 +142,31 @@ def post_install():
         # check whether zookeeper and kafka are running
         ServiceV().validate('isrunning', ['kafka-zookeeper.service', \
             'kafka.service'])
-
+        Log.debug("Validated. Both zookeeper and kafka services are running.")
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:145]: Please remove this log. No need of excessive logging.

File: py-utils/src/setup/utils.py [hunk 27]
```diff
@@ -136,20 +142,31 @@ def post_install():
...
         # Check python packages and install if something is missing
         cmd = "pip3 freeze"
         cmd_proc = SimpleProcess(cmd)
         stdout, stderr, rc = cmd_proc.run()
-        result = stdout.decode('utf-8') if rc == 0 else \
-            stderr.decode('utf-8')
+        if rc == 0:
+            Log.debug(f"cmd {cmd} executed, List of installed packages "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:151]: Refer to v_pkg.py It already has way to check installed pip3 package, please use that.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 27 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py:151]: In v_pkg.py there is 'pip3 list' cmd usage and we are using requirements.txt (pip3 freeze)to verify , they do not match

pip3 list -> aiohttp (3.6.1)
pip3 freeze -> aiohttp==3.6.1

File: py-utils/src/setup/utils.py (hunk 25 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py:137]: We will not be having multiple values in cluster.conf similar to message_bus.conf
{
  "server_node": {
    "site_id": 123,
    "rack_id": 456,
    "node_id": 789
  }
}

File: py-utils/src/utils/log.py [hunk 28]
```diff
@@ -80,7 +82,8 @@ def _get_logger(syslog_server: str, syslog_port: str, file_name: str, logger_typ
                                   maxBytes=max_bytes, backupCount=backup_count)
             file_handler.setFormatter(formatter)
             logger.setLevel(log_level)
-            logger.addHandler(file_handler)
+            if not logger.hasHandlers():
```
Comment (vastradparayya) [inline py-utils/src/utils/log.py:85]: For removal of duplicate entry in log.

File: py-utils/src/setup/utils.py (hunk 25 above)
Comment (ujjwalpl) [reply py-utils/src/setup/utils.py:137]: Today we dont. We will have several values tomorrow. We need to write generic logic.

File: py-utils/src/setup/utils_setup.py [hunk 29]
```diff
@@ -166,15 +168,22 @@ def process(self):
...
+        Log.info(f"Starting utils_setup {argv[1]} ")
         command = Cmd.get_command(desc, argv[1:])
         rc = command.process()
         if rc != 0:
-            raise ValueError(f"Failed to run {argv[1]}")
-    except Exception as e:
+            Log.info(f"Command {command} {argv[1]} finished with exit " \
+                f"code {rc}")
+            sys.exit(rc)
+        Log.info(f"Command {command} {argv[1]} finished with exit " \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:178]: Not only non exception but finished log should be printed in all the cases. 
Move this line to the end after exception. 
try: 
    log.info(starting...)
    xxx
excpt:
    yyy
log.info(Command finished with exit code ...)

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 25 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py:137]: @ujjwalpl This is node specific as per format . Can you give me a example so I can understand more clearly.



cortx:  software:    common:      message_bus_type: kafka # kafka is the only supported type    kafka:      servers:      # Value should be in format &lt;FQDN&gt;:&lt;port&gt;. Port is optional      - TMPL_KAFKA_SERVER1_FQDN:TMPL_KAFKA_SERVER1_PORT      - TMPL_KAFKA_SERVER2_FQDN:TMPL_KAFKA_SERVER2_PORT      - TMPL_KAFKA_SERVER3_FQDN:TMPL_KAFKA_SERVER3_PORTserver_node:  TMPL_MACHINE_ID:    site_id: TMPL_SITE_ID    rack_id: TMPL_RACK_ID    node_id: TMPL_NODE_ID

File: py-utils/src/setup/utils_setup.py (hunk 29 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py [hunk 30]
```diff
@@ -166,15 +168,22 @@ def process(self):
 def main(argv: dict):
     try:
         desc = "CORTX Utils Setup command"
+        Log.info(f"Starting utils_setup {argv[1]} ")
         command = Cmd.get_command(desc, argv[1:])
         rc = command.process()
         if rc != 0:
-            raise ValueError(f"Failed to run {argv[1]}")
-    except Exception as e:
+            Log.info(f"Command {command} {argv[1]} finished with exit " \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:175]: No need of this check and log. Last line already covers this. 
```
    ...
        rc = command.process()

    except SetupError as e:
        sys.stderr.write("error: %s\n\n" % str(e))
        sys.stderr.write("%s\n" % traceback.format_exc())
        Cmd.usage(argv[0])
    except Exception as e:
        sys.stderr.write("error: %s\n\n" % str(e))
        sys.stderr.write("%s\n" % traceback.format_exc())
    
    Log.info(f"Command {command} {argv[1]} finished with exit " \
        f"code {rc}")
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 31]
```diff
@@ -194,14 +214,15 @@ def test():
         # Recieve the same & validate
         msg = msg_test.receive_msg()
         if str(msg.decode('utf-8')) != "Test Message":
-            raise SetupError(errno.EINVAL, "Unable to test the config. Received \
+            Log.error("Could not test utils config, fails in matching sent " +\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:217]: Please use the same message as the exception.

File: py-utils/src/setup/utils.py [hunk 32]
```diff
@@ -44,36 +50,48 @@ class Utils:
...
+            Conf.set('index', 'message_broker>type', 'kafka')
+            servers = [{'server': kafka_server_list[i], 'port': port_list[i]}
+                       for i in range(len(kafka_server_list)) if
+                       {'server': kafka_server_list[i],
+                        'port': port_list[i]} not in pre_servers]
+            servers.extend(pre_servers)
+            Conf.set('index', 'message_broker>cluster', servers)
+            Conf.save('index')
+        except ConfError as e:
+            SetupError(e.rc, "Unable to load servers from config file %s %s",\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:71]: raise ?

File: py-utils/src/setup/utils.py [hunk 33]
```diff
@@ -44,36 +50,48 @@ class Utils:
...
+                conf_file, e)
+        except Exception as e:
+            SetupError(errors.ERR_OP_FAILED, "Unable to load servers from "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:74]: raise ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 33 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 32 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 31 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 30 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils.py (hunk 27 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 26 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 25 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 24 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 23 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 22 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 21 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 20 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 19 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 18 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 17 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 16 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 15 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 14 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 13 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 12 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 11 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 10 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 9 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 8 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 7 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 6 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 5 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 4 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py]: Done

File: py-utils/src/setup/utils.py (hunk 3 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py [hunk 34]
```diff
@@ -44,36 +50,35 @@ class Utils:
...
         Conf.set('index', 'message_broker>type', 'kafka')
         for i in range(len(kafka_server_list)):
             Conf.set('index', f'message_broker>cluster[{i}]', \
                 {'server': kafka_server_list[i], 'port': port_list[i]})
         Conf.save('index')
         # copy this conf file as message_bus.conf
         try:
-            os.rename('/etc/cortx/message_bus.conf.sample', \
-                      '/etc/cortx/message_bus.conf')
+            os.rename(f'/{conf_file}.sample', conf_file)
```
[the remark this answers is not in the document]
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py [hunk 35]
```diff
@@ -15,14 +15,19 @@
...
 from cortx.utils.process import SimpleProcess
-from cortx.utils.validator.v_confkeys import ConfKeysV
 from cortx.utils.validator.v_service import ServiceV
+from cortx.utils.validator.v_confkeys import ConfKeysV
 from cortx.utils.message_bus.error import MessageBusError
-from cortx.utils import errors
+
+
+Log.init('utils_setup', '/var/log/cortx/utils', level='INFO', backup_count=5,\
+         file_size_in_mb=5, syslog_server=None, syslog_port=None)
```
[the remark this answers is not in the document]
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 2 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py (hunk 1 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py [hunk 36]
```diff
@@ -23,6 +23,10 @@
 from cortx.utils.validator.v_service import ServiceV
 from cortx.utils.message_bus.error import MessageBusError
 from cortx.utils import errors
+from cortx.utils.log import Log
+
+Log.init('utils', '/var/log/cortx/utils', level='INFO', backup_count=100,\
```
[the remark this answers is not in the document]
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils.py [hunk 37]
```diff
@@ -44,36 +50,48 @@ class Utils:
...
+                pre_servers = Conf.get('pre_index', 'message_broker>cluster')
+            Conf.load('index', f'json://{conf_file}.sample')
+            Conf.set('index', 'message_broker>type', 'kafka')
+            servers = [{'server': kafka_server_list[i], 'port': port_list[i]}
+                       for i in range(len(kafka_server_list)) if
+                       {'server': kafka_server_list[i],
+                        'port': port_list[i]} not in pre_servers]
+            servers.extend(pre_servers)
+            Conf.set('index', 'message_broker>cluster', servers)
+            Conf.save('index')
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:69]: You are updating sample file. You should update original conf file.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils_setup.py [hunk 38]
```diff
@@ -15,17 +15,21 @@
...
-import argparse
 import inspect
+import argparse
 import traceback
 
 from cortx.setup import Utils
+from cortx.utils.log import Log
+from cortx.setup.utils import SetupError
+
+Log.init('utils_setup', '/var/log/cortx/utils', level='INFO', backup_count=5,\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils_setup.py:26]: Log.init should be called only once from the main program.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 39]
```diff
@@ -63,17 +69,16 @@ def _create_msg_bus_config(kafka_server_list: list, port_list: list):
...
-                    'cortx>software>kafka>servers']
+        key_list = ['cortx>software>common>message_bus_type',
+                   'cortx>software>kafka>servers']
         ConfKeysV().validate('exists', 'cluster_config', key_list)
-
         msg_bus_type = Conf.get('cluster_config', key_list[0])
         if msg_bus_type != 'kafka':
-            raise SetupError(errno.EINVAL, "Message Bus do not support type \
-                %s", msg_bus_type)
+            Log.error(f"Message Bus do not support message bus type "\
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:78]: Log.error(f"Message bus type {msg_bus_type} is not supported")

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 39 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils.py]: Done

File: py-utils/src/setup/utils_setup.py (hunk 38 above)
Comment (vastradparayya) [reply py-utils/src/setup/utils_setup.py:27]: Done

**Review by ujjwalpl — APPROVED** [review]

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Review by selvakumaar5496 — APPROVED** [review]

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**